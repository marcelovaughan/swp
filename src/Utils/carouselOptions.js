export const carouselOptions = { 
    focusAt: "center",
    type: "slide",
    startAt: 0,  
    perView: 3,
    peek: 50,
    gap: 30,
    autoplay: false,
    hoverpause: !1,
    animationDuration: 2e3,
    rewindDuration: 2e3,
    perTouch: 0,
    breakpoints: {
        480: {
            gap: 5,
            peek: 20,
            perView: 1
        },
        768: {
            perView: 2
        },
        1360: {
            perView: 3
        },
        1600: {
            perView: 4
        },
        1960: {
            perView: 3
        }
    }
  };