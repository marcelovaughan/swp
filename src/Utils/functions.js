export function getRandomInt(min, max) {
    min = Math.ceil(min);
    max = Math.floor(max);
    return Math.floor(Math.random() * (max - min + 1)) + min;
}

export function formatNumber(number){
    return number.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
}

export const sleep = timeout => {
    return new Promise(resolve => {
        setTimeout(() => {
            resolve();
        }, timeout);
    });
};

export const getFilmsCounterText = (films) => {
    let legend = ''
    
    if(films.length === 0){
        legend = '(citation only)'
    }else{
        legend = films.length > 1 ? 'films' : 'film'
    }
    return `${films.length + ` ` + legend }`
}