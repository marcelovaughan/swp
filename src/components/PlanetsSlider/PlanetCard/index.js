import React, { useEffect } from 'react';
import PropTypes from "prop-types";

import { formatNumber, getFilmsCounterText } from '../../../Utils/functions';
import Loader from '../../Loader'


export default function PlanetCard(props) {
    
    const { index, isActive, planet, loading, handlegGetPlanet } = props

    useEffect(() => {    
        if(isActive){
            setTimeout(()=>{
                console.log('chamou')
                handlegGetPlanet(index+1)
            }, 2500)
        }           
    }, [handlegGetPlanet, index, isActive])

    return (
        <li className="frames__item glide__slide">
            <div className="frame" data-ref="planet[el]">
                <div className="notch-contain notch-top"></div>
                <React.Fragment>
                    { loading && !planet && <Loader/>  }
                    <div className="content-contain">
                    {planet && !loading &&
                        <React.Fragment>
                            <h2 data-test="name">{planet && planet.name && planet.name}</h2>

                            <p className="planet-attribute">
                                <span className="font-bold">Population: </span>
                                <span className="font-light" data-test="population">{planet && planet.population && formatNumber(planet.population)}</span>
                            </p>
                            <p className="planet-attribute">
                                <span className="font-bold">Climate: </span>
                                <span className="font-light" data-test="climate">{planet && planet.climate && planet.climate}</span>
                            </p>
                            <p className="planet-attribute">
                                <span className="font-bold">Terrain: </span>
                                <span className="font-light" data-test="terrain">{planet && planet.terrain && planet.terrain}</span>
                            </p>
                            <p className="planet-attribute">
                                <span className="font-bold">Featured in: </span>
                                <span className="font-light" data-test="films">
                                    {planet && planet.films && `${ getFilmsCounterText(planet.films) }`}                                    
                                </span>
                            </p>                            
                        </React.Fragment>                        
                    }
                    </div>
                </React.Fragment>
                <div className="bottom-notch-contain"> 
                    <div className="notch-left"></div>
                    <div className="notch-right"></div> 
                </div>  
            </div>
        </li>
    );
}

PlanetCard.defaulProps = {
    loading: false, 
}

PlanetCard.propTypes = {
    index: PropTypes.number, 
    isActive: PropTypes.bool, 
    planet: PropTypes.object, 
    loading: PropTypes.bool, 
    handlegGetPlanet: PropTypes.func    
};

