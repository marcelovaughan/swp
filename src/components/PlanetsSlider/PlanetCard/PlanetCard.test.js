import React from "react"
import { configure, mount, shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import PlanetCard from "./index";

import { formatNumber, getFilmsCounterText } from '../../../Utils/functions';

const mockedPlanet = 
    { name: 'Tatooine',
    climate: 'arid',
    terrain: 'desert',
    population: '200000',
    films:
    [   'https://swapi.co/api/films/5/',
        'https://swapi.co/api/films/4/',
        'https://swapi.co/api/films/6/',
        'https://swapi.co/api/films/3/',
        'https://swapi.co/api/films/1/' ],
    }

configure({ adapter: new Adapter() });

describe("Planet Card Render", () => {
  
    jest.setTimeout(120000);
  
    const defaultProps = {
        index: null, 
        isActive: false, 
        planet: null, 
        loading: false, 
        handlegGetPlanet: ()=>{}
    };

    const successProps = {
        index: 0, 
        isActive: true, 
        planet: mockedPlanet, 
        loading: false, 
        handlegGetPlanet: ()=>{}
    };
 

    it("Render the component", () => {
        const component = mount(<PlanetCard {...defaultProps} />);
        expect(component.debug()).toMatchSnapshot();
    });

    it("Render the component with Tatooine Planet info", async () => {        
        const wrapper = shallow(<PlanetCard {...successProps} />);

        const name = wrapper.find('[data-test="name"]').children().text();
        expect(name).toBe(mockedPlanet.name);

        const population = wrapper.find('[data-test="population"]').children().text();
        expect(population).not.toBe(mockedPlanet.population);        
        expect(population).toBe(formatNumber(mockedPlanet.population));

        const climate = wrapper.find('[data-test="climate"]').children().text();
        expect(climate).toBe(mockedPlanet.climate);

        const films = wrapper.find('[data-test="films"]').children().text();
        const filmsText = `${getFilmsCounterText(mockedPlanet.films) }`
        expect(films).toBe(filmsText);        
    });

    it("Render the component with Zero featured films", async () => {        
        mockedPlanet.films = []
        const wrapper = shallow(<PlanetCard {...successProps} />);
        const films = wrapper.find('[data-test="films"]').children().text();        

        const filmsTextEmpty = "0 (citation only)"
        expect(films).toBe(filmsTextEmpty);
    });
});
