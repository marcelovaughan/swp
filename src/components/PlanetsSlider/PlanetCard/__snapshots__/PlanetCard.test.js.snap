// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet Card Render Render the component 1`] = `
"<PlanetCard index={{...}} isActive={false} planet={{...}} loading={false} handlegGetPlanet={[Function: handlegGetPlanet]}>
  <li className=\\"frames__item glide__slide\\">
    <div className=\\"frame\\" data-ref=\\"planet[el]\\">
      <div className=\\"notch-contain notch-top\\" />
      <div className=\\"content-contain\\" />
      <div className=\\"bottom-notch-contain\\">
        <div className=\\"notch-left\\" />
        <div className=\\"notch-right\\" />
      </div>
    </div>
  </li>
</PlanetCard>"
`;
